### PR TITLE
Better upgrade visibility in editor UI

### PR DESCRIPTION
**Pull request by Oliveriver**
**Brief Description of What This PR Does**

* Displays upgrade icon in buttons for organelles that can be modified/upgraded
* Displays modification notice in tooltips for organelles that can be modified/upgraded
* Automatically opens the organelle modify/upgrade UI when a modifiable organelle is placed, under the following conditions:
    * Organelle was not placed with symmetry
    * Organelle placement tutorial is not active

Note this often results in the player having to undo twice if they want to remove a modifiable organelle due to #4091 which sadly still seems to be a bug.

image

**Related Issues**

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [ ] Initial code review passed (this and further items should not be checked by the PR author)
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

File: src/microbe_stage/editor/CellEditorComponent.cs [hunk 1]
```diff
@@ -1649,6 +1645,14 @@ private bool AddOrganelle(string organelleType)
                     {
                         usedHexes.Add(hex);
                     }
+
+                    if (organelleDefinition.CanBeModified &&
+                        Symmetry == HexEditorSymmetry.None &&
+                        (TutorialState == null || !TutorialState.Enabled || TutorialState.EditorUndoTutorial.Complete))
```
Comment (Oliveriver) [inline src/microbe_stage/editor/CellEditorComponent.cs:1651]: This feels like a pretty ugly way to check the tutorial state is safe. Is there a better way to do it that I should be aware of? Or should this section be placed somewhere else entirely to avoid it?

**Comment by hhyyrylainen** [comment]
> Automatically opens the organelle modify/upgrade UI when a modifiable organelle is placed, under the following conditions: 

I disagree with this being the default. I think fundamentally players should be fine with playing default organelle state for a while, and upgrades are mostly something different. So prompting immediately to modify a placed organelle doesn't make sense in a lot of cases. For example vacuoles, I don't think players would always want to specialize them, or for the pilus to convert it to deal toxin damage.

There could be an option in the options menu to enable this but I don't think it should be on by default.

File: src/microbe_stage/editor/CellEditorComponent.cs (hunk 1 above)
Comment (hhyyrylainen) [reply src/microbe_stage/editor/CellEditorComponent.cs:1651]: There is `TutorialState.TutorialActive` to see if any tutorial is currently open.

**Comment by Oliveriver** [comment]
Fair enough. Would automatically opening the right click menu be a sensible compromise? Or should I just change the text in the tooltip to read 'Right click to modify once placed' and leave it as it is otherwise?

**Comment by hhyyrylainen** [comment]
I'm not sure the menu opening would be that good flow either. At least it would require me extra clicks when building specific things to test. Also explicitly referring to right click is not a long term solution (I suggest using as neutral terminology as possible to avoid having a separate text for controller input): https://github.com/Revolutionary-Games/Thrive/issues/4003

**Comment by Oliveriver** [comment]
Right, but I'm just concerned that it's still obscuring information from the player if they aren't shown or told explicity how to modify the organelle. I'll remove the menu auto-opening for now so the tooltip and icon changes can at least go in.

File: src/microbe_stage/editor/CellEditorComponent.cs (hunk 1 above)
Comment (Oliveriver) [reply src/microbe_stage/editor/CellEditorComponent.cs]: Removed anyway now so no longer relevant,

File: src/microbe_stage/editor/CellEditorComponent.GUI.cs [hunk 2]
```diff
@@ -260,6 +260,7 @@ private void UpdateOrganelleUnlockTooltips()
             var tooltip = GetSelectionTooltip(organelle.InternalName, "organelleSelection");
             if (tooltip != null)
             {
+                tooltip.CanBeModified = organelle.CanBeModified;
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/CellEditorComponent.GUI.cs:263]: Is this change actually required considering the code in `SetupMicrobePartSelections`?
If this is needed, then I think it might warrant renaming this method to something like `UpdateOrganelleTooltips` (as long as there isn't already a method with basically that name).

File: locale/en.po [hunk 3]
```diff
@@ -3892,6 +3892,10 @@ msgstr "Models"
 msgid "MODIFY"
 msgstr "Modify"
 
+#: ../src/microbe_stage/editor/tooltips/SelectionMenuToolTip.tscn:163
+msgid "MODIFY_ONCE_PLACED"
+msgstr "Modify once placed"
```
Comment (hhyyrylainen) [inline locale/en.po:3897]: Is it the best that this text tells the player "hey, modify this" instead of being like "Can be modified after placing"? I feel like that's a bit off in terms of tone in a general characteristics info box listing.

File: src/microbe_stage/editor/tooltips/SelectionMenuToolTip.tscn [hunk 4]
```diff
@@ -138,14 +140,36 @@ margin_right = 333.0
...
+expand = true
+stretch_mode = 6
+
+[node name="CanBeModifiedLabel" type="Label" parent="MarginContainer/VBoxContainer/CanBeModified"]
+margin_left = 20.0
+margin_right = 192.0
+margin_bottom = 19.0
+custom_colors/font_color = Color( 1, 0.898039, 0.337255, 1 )
+custom_fonts/font = ExtResource( 10 )
+text = "MODIFY_ONCE_PLACED"
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/tooltips/SelectionMenuToolTip.tscn:163]: I think the text key would be better to have something like "PART_IS_MODIFIABLE"

File: src/microbe_stage/editor/CellEditorComponent.cs [hunk 5]
```diff
@@ -2015,6 +2010,7 @@ private void SetupMicrobePartSelections()
             var control = (MicrobePartSelection)organelleSelectionButtonScene.Instance();
             control.Locked = organelle.Unimplemented;
             control.PartIcon = organelle.LoadedIcon ?? throw new Exception("Organelle with no icon");
+            control.CanBeModified = organelle.CanBeModified;
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/CellEditorComponent.cs:2013]: Ah yeah, now I remember that this was separate from the tooltip. So this is also needed.
I guess we decided that for now showing the modifiable icon on the organelles that can be modified is fine. It'll be easy enough to make this an option / disable it once almost every single organelle can be modified.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
Besides the translation conflict, I'd like to see a tiny bit of technical side clean up for this before merging.
